core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -55,46 +55,51 @@ func getCommitMethodConfig(fromAddress string, routerProgramAddress string, comm
 				IsWritable: true,
 			},
 			chainwriter.PDALookups{
-				Name: "RouterReportAccount",
-				PublicKey: chainwriter.AccountConstant{
-					Address:    routerProgramAddress,
-					IsSigner:   false,
-					IsWritable: false,
-				},
+				Name:      "CommitReport",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
 				Seeds: []chainwriter.Seed{
 					{Static: []byte("commit_report")},
 					{Dynamic: chainwriter.AccountLookup{Location: merkleRootChainSelector}},
-					{Dynamic: chainwriter.AccountLookup{
-						Location: "Info.MerkleRoots.MerkleRoot",
-					}},
+					{Dynamic: chainwriter.AccountLookup{Location: merkleRoot}},
 				},
 				IsSigner:   false,
-				IsWritable: false,
+				IsWritable: true,
 			},
 			getAuthorityAccountConstant(fromAddress),
 			getSystemProgramConstant(),
-			chainwriter.AccountConstant{
-				Name:       "SysvarInstructions",
-				Address:    sysvarInstructionsAddress,
-				IsSigner:   true,
+			getSysVarInstructionConstant(),
+			chainwriter.PDALookups{
+				Name:      "FeeBillingSigner",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("fee_billing_signer")},
+				},
+				IsSigner:   false,
 				IsWritable: false,
 			},
+			getFeeQuoterConfig(offrampProgramAddress),
```

Comment:
is this expected ? I see it's called again in line 84

Reply:
Ya this kind of looks weird but it's expected. It's to cover these 2 accounts for in the [CommitReportContext](https://github.com/smartcontractkit/chainlink-ccip/pull/553/files#diff-93da077ce07b35cc91a145120ac2ca90ade6a5785bd3244d6af5aac10d25cce3R320-R331). We have to provide the `FeeQuoter` address as one of the accounts which is this reference. But we also have to provide the `FeeQuoterConfig` account which is a PDA on the `FeeQuoter`. So unlike the other PDAs where we use the offramp address, we have to fetch the `FeeQuoter` address to use as the public key for the reference on line 84